File: Strings/4/backspaceString.py
def backspaceCompare(S,T):
    firstString, secondString = "", ""
    for ele in S:
        if ele == '#':
            firstString = firstString[:len(firstString)-1]
        else:
            firstString += ele
    for ele in T:
        if ele == '#':
            secondString = secondString[:len(secondString)-1]
        else:
            secondString += ele
    return firstString == secondString

print(backspaceCompare(S="ab##", T=""))

File: Strings/4/backspaceString.js
var backspaceCompare = function (S, T) {
  let firstString = "", secondString = "";
  for (let i=0; i<S.length; i++) {
    if (S[i] === '#') {
      firstString = firstString.slice(0, firstString.length-1);
    }
    else {
      firstString += S[i];
    }
  }
  for (let i=0; i<T.length; i++) {
    if (T[i] === '#') {
      secondString = secondString.slice(0, secondString.length-1);
    }
    else {
      secondString += T[i];
    }
  }
  if (firstString === secondString) {
    return true
  }
  else {
    return false
  }
};

console.log(backspaceCompare(S="ab##", T=""))